src/Facebook/InstantArticles/Validators/Type.php

```diff
@@ -396,9 +398,10 @@ public static function isTextEmpty($text)
             return true;
         }
         // Stripes empty spaces, &nbsp;, <br/>, new lines
-        $text = strip_tags($text);
-        $text = preg_replace("/[\r\n\s]+/", "", $text);
+        $text = preg_replace("/\s+/", "", $text);
         $text = str_replace("&nbsp;", "", $text);
+        $text = str_replace("<br>", "", $text);
+        $text = str_replace("<br/>", "", $text);
```

Comment:
Actually, let's use this opportunity to fix a problem with the use of `strip_tags` in this method. Here, if we're looking at the text content of any Element, it is *not* an HTML snippet, and we do convert it to HTML Entities before outputting. We should consider any text content as plain text.

Example:

```php
Paragraph $p1 = new Paragraph();
$p1->appendText("This is how you make line breaks in HTML:");
Paragraph $p2 = new Paragraph();
$p2->appendText("<br>");
$p1->isValid(); // Should return true.
$p2->isValid(); // Should return true.
````

Let's remove these two lines and fix tests accordingly.